adds support for displaying pending transactions

hughy opened the pull request:
the transaction page in the block explorer assumes that each transaction is on at least one block.

if we begin syncing pending transactions from the api-node to the api then the block explorer will display transactions that are not on any block.

- omits the 'Block Info' breadcrumb link if the query params do not include a block id
- displays a 'Pending' badge if the transaction is not on a block
- omits the 'Block Hash' card if the transaction is not on a block

before:
<img width="924" alt="image" src="https://github.com/iron-fish/block-explorer/assets/57735705/e9577123-6921-4f69-995b-b9e92609c9ea">

after:
![image](https://github.com/iron-fish/block-explorer/assets/57735705/f8aa57b6-08c8-44d3-923c-957f14922142)

Resolves IFL-1030

Commit:
adds support for displaying pending transactions

the transaction page in the block explorer assumes that each transaction is on
at least one block.

if we begin syncing pending transactions from the api-node to the api then the
block explorer will display transactions that are not on any block.

- omits the 'Block Info' breadcrumb link if the query params do not include a
  block id
- displays a 'Pending' badge if the transaction is not on a block
- omits the 'Block Hash' card if the transaction is not on a block

danield9tqh commented on pages/transaction/[hash].tsx:
```diff
@@ -236,13 +240,18 @@ const TransactionInfo: FC<TransactionInfoProps> = ({ data, loaded }) => {
     )
   }, [data?.notes?.length, data?.spends?.length])
 
+  const badge =
```
I think showing expired here would be good eventually. We don't sync expiration sequence right now though maybe can do that in a follow up

Commit:
adds support for expired transactions

queries the chain head in the transaction page and compares its sequence to the
transaction expiration

adds a card for transaction expiration

displays an 'Expired' badge for expired transactions

hughy commented:
Updated to support displaying expired transactions: 
<img width="930" alt="image" src="https://github.com/iron-fish/block-explorer/assets/57735705/2f763a1d-5fec-4d5b-9f1d-1fafe724768c">

danield9tqh commented on components/Breadcrumbs/Breadcrumbs.tsx:
```diff
@@ -94,22 +94,24 @@ const resolvePath = (path: string, queryParams: ParsedUrlQuery | null) => {
           key: 'breadcrumb-explorer',
           link: <BreadcrumbLink.Explorer />,
         },
-        {
```
Can we remove `breadcrumb-explorer` if there are no params too?

Commit:
removes the 'All Blocks' breadcrumb

do not include a breadcrumb link to 'All Blocks' if the transaction is pending
or expired
